### PR TITLE
{devel}[SYSTEM] PyCharm v2017.2.3

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**

### CODE REVIEW

**Comment by shahzebsiddiqui** [comment]
@boegel  i dont like @boegelbot  can you please merge request. I'd like to get PyCharm in central repo as I use it regularly and not have to rely on my local branch.

**Comment by jose-d** [comment]
hi, is there any particular reason to use PyCharm from year 2017?

File: easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name = 'PyCharm'
+version = '2017.2.3'
+easyblock = 'Tarball'
+
+homepage = 'https://www.jetbrains.com/pycharm/'
+description = """PyCharm Community Edition: Python IDE for Professional Developers"""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
```
Comment (jose-d) [inline easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb]: I believe dummy:dummy toolchain was deprecated in favor of toolchain "SYSTEM". One can see it in action i.e. in PR #9095.

**Comment by shahzebsiddiqui** [comment]
No I just happened to have this built 2 years ago and i never got time to send PR. PyCharm would be a new software to easybuild collection since there is no previous easyconfig. I will be happy to send another PR for latest version sometime soon.

File: easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb (hunk 1 above)
Comment (shahzebsiddiqui) [reply easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb]: thanks @jose-d  let me fix this

Comment (shahzebsiddiqui) [reply easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb]: made the fix see https://github.com/easybuilders/easybuild-easyconfigs/pull/9100/commits/a9e74faf1a6c25ed50c0a80c7d137d9db652816d

File: easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name = 'PyCharm'
+version = '2017.2.3'
+easyblock = 'Tarball'
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyCharm/PyCharm-2017.2.3.eb]: @shahzebsiddiqui Style nitpicking: `easyblock` line should be on top.

I've fixed this in https://github.com/shahzebsiddiqui/easybuild-easyconfigs/pull/1, if you merge that PR in your branch the PR here will be updated accordingly.

**Comment by boegel** [comment]
> @boegel i dont like @boegelbot can you please merge request. I'd like to get PyCharm in central > repo as I use it regularly and not have to rely on my local branch.

@shahzebsiddiqui I can see that the failing tests over what seems like trivialities can be annoying, but those tests are important to ensure high-quality EasyBuild releases where easyconfigs that are included work and keep working.

There's a minor style issues where the tests don't trip over yet, I've fixed this with a PR to your branch.

Other than that, this is good to go. It's a bit of an atypical software application to support in EasyBuild, but it doesn't hurt either. :)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3126.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/1304745d44ef032ba1e4544b42449a58 for a full test report.

**Comment by shahzebsiddiqui** [comment]
Thanks for the fix @boegel. Its been quite some time since i contributed back. I had some issues setting up the Easybuild Git integration, partly due to my new shell environment (csh) which i am not used to.  Generally used bash, anyhow will use the ``--new-pr`` and ``--check-syntax`` option to review the changes before merging.  I realize now that easyconfig checks are more strict as pose to before.

**Comment by boegel** [comment]
Going in, thanks @shahzebsiddiqui!

**Pull request merged**